### PR TITLE
refactoring method arguments and adding reflecting planes

### PR DESCRIPTION
**Pull request by Shimwell**
This PR changes the code a little.

users must now used the ```export_xml()``` method to create the xml files prior to using ```.simulate()```

This allows for code simplification at the expense of a little extra work for the users

The option to add csg reflecting surfaces to the model is also now possible thanks to @RemDelaporteMathurin

### CODE REVIEW

**Comment by Shimwell** [comment]
Lokos lie ```diff_between_angles``` is undefined on line 132 in openmc_dagmc_wrapper/neutronics_model.py

File: openmc_dagmc_wrapper/neutronics_model.py [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
+from remove_dagmc_tags import remove_tags
 
-from .utils import (create_initial_particles,
-                    extract_points_from_initial_source, plotly_trace,
-                    silently_remove_file)
+from .utils import (
+    create_initial_particles,
+    extract_points_from_initial_source,
+    plotly_trace,
+    silently_remove_file,
```
Comment (RemDelaporteMathurin) [inline openmc_dagmc_wrapper/neutronics_model.py:18]:
```suggestion
    silently_remove_file, diff_between_angles
```

**Comment by RemDelaporteMathurin** [comment]
> Looks lie `diff_between_angles` is undefined on line 132 in openmc_dagmc_wrapper/neutronics_model.py

@Shimwell it seems you forgot to bring this is during the conflicting merge on the other PR

File: openmc_dagmc_wrapper/neutronics_model.py (hunk 1 above)
Comment (Shimwell) [reply openmc_dagmc_wrapper/neutronics_model.py:18]: Thanks Remi, sorry for missing that one

**Comment by Shimwell** [comment]
Great, that last commit fixed the GH CI with install

**Pull request merged**